source/logrepl/cdc.go

```diff
@@ -64,8 +78,22 @@ func NewCDCIterator(ctx context.Context, pool *pgxpool.Pool, c CDCConfig) (*CDCI
 			Msgf("Publication %q already exists.", c.PublicationName)
 	}
 
-	records := make(chan opencdc.Record)
-	handler := NewCDCHandler(internal.NewRelationSet(), c.TableKeys, records, c.WithAvroSchema)
+	// Using a buffered channel here so that the handler can send a batch
+	// to the channel and start building a new batch.
+	// This is useful when the first batch in the channel didn't reach BatchSize (which is sdk.batch.size).
+	// The handler can prepare the next batch, and the CDCIterator can use them
+	// to return the maximum number of records.
+	batchesCh := make(chan []opencdc.Record, 1)
+	handler := NewCDCHandler(
+		ctx,
+		internal.NewRelationSet(),
+		c.TableKeys,
+		batchesCh,
+		c.WithAvroSchema,
+		c.BatchSize,
```

Comment:
I'm wondering if we could have a default value for this one in case it's not set as part of the configuration in `CDCConfig`. @hariso What do you think?

i.e.: This wouldn't be necessary https://github.com/ConduitIO/conduit-connector-postgres/pull/284/files.

Reply:
@raulb which file specifically are you referring to in the link?

Reply:
@hariso LOL, I grabbed the wrong link, sorry about that 🤦‍♂️. I meant this https://github.com/ConduitIO/conduit-connector-postgres/issues/282

Reply:
@hariso I don't seem to find the line I was referring to. Never mind 🤷‍♂️